orttraining/orttraining/python/training/onnxblock/README.md

```diff
@@ -0,0 +1,82 @@
+
+```py
+import onnxruntime.training.onnxblock as onnxblock
+
+class MyTrainingModel(onnxblock.TrainingModel):
+    def __init__(self):
+        super(MyTrainingModel, self).__init__()
+        self._loss1 = onnxblock.loss.MSELoss()
+        self._loss2 = onnxblock.loss.MSELoss()
+        self._w1 = onnxblock.building_blocks.Constant(0.6)
+        self._w2 = onnxblock.building_blocks.Constant(0.4)
+        self._add = onnxblock.building_blocks.Add()
+        self._mul = onnxblock.building_blocks.Mul()
+
+    def build(self, loss_input_name1, loss_input_name2):
+        # Returns weighted average of the two losses
+        return self._add(
+            self._mul(self._w1(), self._loss1(loss_input_name1, target_name="target1")),
+            self._mul(self._w2(), self._loss2(loss_input_name2, target_name="target2"))
+        )
+
+# Load the model
+model_path = "model.onnx"
+model = onnx.load(model_path)
+
+# Build the training model by adding the loss block and building the
+# gradient graph. This is achieved by invoking an instance of
+# the onnxblock.TrainingModel.
+my_model = MyTrainingModel()
+with onnxblock.onnx_model(model) as accessor:
+    # Invoking the model builder will add the loss block on top
+    # of the loaded onnx_model inplace.
+    loss_output_name = my_model("output-0", "output-1")
+    # Eval model is obtained as a by-product of building
+    # the training model.
+    eval_model = accessor.eval_model
+
+# Save the models
+output_model_path = "training_model.onnx"
+output_eval_model_path = "eval_model.onnx"
+onnx.save(model, output_model_path)
+onnx.save(eval_model, output_eval_model_path)
+```
+
+### The Optimizer Graph
+
+The optimizer graph can be generated in a similar way. An example snippet is as follows:
+
+```py
+cli_grad_norm = onnxblock.optim.ClipGradNorm(2.5)
+optimizer = onnxblock.optim.AdamW(clip_grad=clip_grad_norm)
+optimizer_model = None
+with onnxblock.onnx_model() as accessor:
+    optimizer_outputs = optimizer(my_model.parameters())
+    optimizer_model = accessor.model
+
+# save the optimizer model
+output_optimizer_model_path = "adamw_optimizer.onnx"
+onnx.save(optimizer_model, output_optimizer_model_path)
+```
+
+### The Checkpoint
+
+The model checkpoint can be saved using instance of the `TrainingModel`'s `parameters()` method.
+
+```py
+# save checkpoint
+output_checkpoint_path = "checkpoint.ckpt"
+onnxblock.save_checkpoint(my_model.parameters(), output_checkpoint_path)
+```
+
+Once the models and checkpoint have been generated, they can be loaded in the online training step and executed.
```

Comment:
Add a link to the trainer.cc file which demonstrates C API usage to initiate training session using these artifacts.